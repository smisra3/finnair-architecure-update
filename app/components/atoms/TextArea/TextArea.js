// @flow
import type { Node } from 'react';
import styled from 'styled-components';

import styles from './TextArea.style';


type Props = {
  className: string,
  id: string | number,
  labelContent?: string,
  iconClass?: string,
  inheritedStyles?: string,
  isError?: boolean,
  errorMessage: string,
  hideLabel?: boolean,
};

const TextArea = ({
  className,
  id,
  labelContent,
  iconClass,
  isError,
  errorMessage,
  hideLabel,
  inheritedStyles,
  ...others
}: Props): Node => (
  <div className={className}>
    <label htmlFor={id}>
      <span className={`${hideLabel ? 'is-hidden' : ''}`}>{labelContent}</span>

      {iconClass ? <i className={iconClass} /> : null}
      <textarea data-error={isError} id={id} {...others} />
    </label>
    {isError && errorMessage ? <p className="error">{errorMessage}</p> : null}
  </div>
);

TextArea.defaultProps = {
  labelContent: null,
  iconClass: '',
  inheritedStyles: '',
  isError: false,
  hideLabel: false,
};

export default styled(TextArea)`${styles}`;
export { TextArea as TextAreaComponent };
