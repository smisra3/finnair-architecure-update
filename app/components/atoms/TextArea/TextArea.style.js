import { css } from 'styled-components';

export default css`
  label {
    width: 100%;
    display: flex;
  }

  textarea {
    border: none;
    width: 0;
    flex: 1 1;
    height: 200px;

    &::placeholder {
      color: ${props => props.theme.inputPlaceholder};
    }

    &:focus {
      outline: none;
    }
  }

  .error {
    font-family: ${props => props.theme.secondaryFontFamily};
    font-size: 10px;
    letter-spacing: 1px;
    text-align: left;
    color: ${props => props.theme.errorColor};
    margin-top: 5px;
  }
`;
