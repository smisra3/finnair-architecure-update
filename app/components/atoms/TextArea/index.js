// @flow

/**
 * Created by dcha17 on 1/15/2018.
 */
export { default } from './TextArea';
