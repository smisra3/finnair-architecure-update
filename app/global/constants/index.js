/* TODO: Remove lint exception */
/* eslint-disable import/prefer-default-export */
export const DEVICE_TYPE = 'DEVICE_TYPE';
export const IS_TABLET = 'IS_TABLET';
export const REGION_TYPE = 'REGION_TYPE';
export const CONFIG_KEYS = 'CONFIG_KEYS';
export const CURRENT_ROUTE = 'CURRENT_ROUTE';
export const PAGE_URL = 'PAGE_URL';
export const PAGE_QUERY = 'PAGE_QUERY';
export const GLOBAL_DATA_FAILURE = 'FIL/GLOBAL/GLOBAL_DATA_FAILURE';
export const CHECK_AUTH = 'FIL/GLOBAL/CHECK_AUTH';
export const CHECK_AUTH_SUCCESS = 'FIL/GLOBAL/CHECK_AUTH_SUCCESS';
export const SET_USER_STATE = 'FIL/GLOBAL/SET_USER_STATE';
export const PAGE_ORIGIN = 'PAGE_ORIGIN';
export const LABELS_STATE_KEY = 'labels';

export const USER_STATE_LOGGED_IN = 'FIL/GLOBAL/SUCCESS';
export const USER_STATE_RECOGNIZED = 'FIL/GLOBAL/RECOGNIZED';
export const USER_STATE_ANONYMOUS = 'FIL/GLOBAL/ANONYMOUS';

export const GET_LABELS = 'FIL/GLOBAL/GET_LABELS';
export const GET_LABELS_SUCCESS = 'FIL/GLOBAL/GET_LABELS_SUCCESS';
export const GET_LABELS_FAILURE = 'FIL/GLOBAL/GET_LABELS_FAILURE';
