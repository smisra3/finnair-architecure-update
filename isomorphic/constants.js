const HEARTBEAT_SESSION_NAME = 'JSESSIONID';
const ENV_DEVELOPMENT = 'development';
const ENV_PRODUCTION = 'production';
const SERVICE_WORKER_FILE = '/service-worker.js';
const API_PROXY_PATH = '/api';
const CLEAR_SERVICES_CACHE = '/clear-services-cache';
const HEALTH_CHECK = '/healthcheck';
const PROTOCOL_HTTPS = 'https';
const RESPONSE_OK = 200;
const RESPONSE_INTERNAL_SERVER_ERROR = 500;
const STATIC_FOLDER_PREFIX = '/static/';

module.exports = {
  HEARTBEAT_SESSION_NAME,
  ENV_DEVELOPMENT,
  ENV_PRODUCTION,
  SERVICE_WORKER_FILE,
  API_PROXY_PATH,
  CLEAR_SERVICES_CACHE,
  PROTOCOL_HTTPS,
  RESPONSE_OK,
  RESPONSE_INTERNAL_SERVER_ERROR,
  STATIC_FOLDER_PREFIX,
  HEALTH_CHECK,
};
