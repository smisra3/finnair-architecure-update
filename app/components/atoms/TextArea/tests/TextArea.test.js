import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { TextAreaComponent } from '../TextArea';
import TextAreaStyle from '../';

import Theme from '../../../../styles/theme';
import { DESKTOP } from '../../../../constants';

const theme = {
  ...Theme,
  deviceType: DESKTOP,
};

const props = {
  className: '',
  id: '1',
  labelContent: '',
  iconClass: '',
  isError: false,
  errorMessage: 'error',
  hideLabel: false,
  inheritedStyles: '',
};

describe('PaymentMethod component', () => {
  let textAreaComponent = '';
  test('should render correctly', () => {
    textAreaComponent = shallow(<TextAreaComponent {...props} />);
    expect(textAreaComponent).toMatchSnapshot();
  });
});

describe('PaymentMethodStyle component', () => {
  let textAreaComponent = '';
  test('should render correctly', () => {
    textAreaComponent = shallow(<TextAreaStyle {...props} {...theme} />);
    expect(textAreaComponent).toMatchSnapshot();
  });
});
