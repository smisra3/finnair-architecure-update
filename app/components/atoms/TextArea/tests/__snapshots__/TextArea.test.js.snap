// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethod component should render correctly 1`] = `
<div
  className=""
>
  <label
    htmlFor="1"
  >
    <span
      className=""
    />
    <textarea
      data-error={false}
      id="1"
    />
  </label>
</div>
`;

exports[`PaymentMethodStyle component should render correctly 1`] = `
.c0 label {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 textarea {
  border: none;
  width: 0;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  height: 200px;
}

.c0 textarea:focus {
  outline: none;
}

.c0 textarea:focus {
  outline: none;
}

.c0 textarea:focus {
  outline: none;
}

.c0 textarea:focus {
  outline: none;
}

.c0 .error {
  font-size: 10px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-align: left;
  margin-top: 5px;
}

<TextArea
  backgroundColorDisabled="grey"
  backgroundColorPrimary="#fff"
  backgroundColorSecondary="#eaf2f6"
  backgroundColorSuccess="#2e993a"
  brandPrimaryColor="#006193"
  buttonBackgroundColorPrimary="#e57725"
  buttonBackgroundColorSecondary="#016fad"
  buttonColorPrimary="#ffffff"
  buttonColorSecondary="#ffffff"
  cautionBackgroundColor="#fffdf3"
  className="c0"
  colorOutline="#999"
  containerShadowPrimary="rgba(0,0,0,.2)"
  deviceType="desktop"
  errorMessage="error"
  fieldBorderColor="#bfbfbf"
  fieldBorderDisabledColor="#bfdfdf"
  fieldBorderErrorColor="#c00"
  fieldBorderFocusColor="#006193"
  fontBold="700"
  fontBook="400"
  fontFamilyDefault="Arial, Helvetica, sans-serif"
  fontFamilyIcon=""
  fontFamilyPrimary="Arial, Helvetica, sans-serif"
  fontFamilySecondary="Arial, Helvetica, sans-serif"
  fontMedium="500"
  fontNormal="normal"
  fontSemiBold="600"
  fontSizeBase="16px"
  fontSizeButtonPrimary="24px"
  fontSizeButtonSecondary="16px"
  fontSizeDefault="12px"
  hideLabel={false}
  iconClass=""
  id="1"
  inheritedStyles=""
  isError={false}
  labelContent=""
  letterSpacing="1px"
  lineHeightBase="1.5"
  linkColorPrimary="#008CBA"
  textColorError="#c00"
  textColorPlaceholder="#dfdfdf"
  textColorPrimary="#222"
  textColorSecondary="#262626"
  textColorSuccess="#2e993a"
  zIndex={
    Object {
      "zFilter": 100,
      "zFilterActions": 400,
      "zFilterList": 150,
      "zFlyout": 200,
      "zLoader": 1000,
      "zMenu": 800,
      "zNativeTickSelectIcon": 50,
      "zOverlay": 900,
      "zPlacesAutocomplete": 1100,
      "zSticky": 700,
    }
  }
/>
`;
